pkg/logger/zap.go

```diff
@@ -2,45 +2,62 @@ package logger
 
 import (
 	"fmt"
+	"os"
 	"time"
 
 	"github.com/go-logr/zapr"
+	"github.com/pkg/errors"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
 )
 
+// InitZapArgs represents a set of arguments for initializing the zap logger.
+type InitZapArgs struct {
+	Level          int
+	OutputFilePath string
+	Opts           []LoggerOpt
```

Comment:
This seems a bit too complex, nested options in options. I would just make this another normal field, like `WithNames []string`.